### PR TITLE
Bug 565660 workbench restriction executor

### PR DESCRIPTION
**Pull request by eparovyshnaya**
Implement error viewing facility for Diagnostic Dialog.

Signed-off-by: elena.parovyshnaya <elena.parovyshnaya@gmail.com>

### CODE REVIEW

File: bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/FirstSelected.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	private final Class<T> cls;
+
+	FirstSelected(ISelection selection, Class<T> cls) {
+		this.selection = selection;
+		this.cls = cls;
+	}
+
+	@Override
+	public Optional<T> get() {
+		if (!IStructuredSelection.class.isInstance(selection)) {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/FirstSelected.java:33]: `Optional.ofNullable(selection).filter(IStructuredSelection.class::isInstance).map(IStructuredSelection.class::cast) `
an so on

File: bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/StatusFromException.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.passage.lic.internal.api.diagnostic.Diagnostic;
+
+public final class StatusFromException implements Supplier<IStatus> {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/StatusFromException.java:26]: Please move to lic.equinox

Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/StatusFromException.java:26]: Oops, we face an accident!

Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/StatusFromException.java]: All the work has been moved to `internal` package

File: bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/FirstSelected.java (hunk 1 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/jface/dialogs/licensing/FirstSelected.java]: _IF_s are reworked to _optional_ flow

File: bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/internal/jface/dialogs/licensing/DiagnosticDialog.java [hunk 3]
```diff
@@ -55,38 +43,48 @@ protected void configureShell(Shell shell) {
...
+		new ButtonConfig(2, new CopyToClipboard(this::getShell, new DiagnosticExplained(diagnostic)), "Co&py", "", "")//$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+				.reside(buttons);
 	}
 
 	@Override
-	protected void createButtonsForButtonBar(Composite parent) {
-		createButton(parent, IDialogConstants.CLOSE_ID, IDialogConstants.CLOSE_LABEL, true);
+	protected void buildUI(Composite parent) {
+		viewer = new LicensingTable<Trouble>(parent, Trouble.class) //
+				.withColumn("Error", 720, t -> t.details()) //$NON-NLS-1$
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/internal/jface/dialogs/licensing/DiagnosticDialog.java:56]: translatable

File: bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/internal/jface/dialogs/licensing/DiagnosticDialog.java [hunk 4]
```diff
@@ -55,38 +43,48 @@ protected void configureShell(Shell shell) {
...
+				.withColumn("Code", 50, t -> Integer.toString(t.code().code())) //$NON-NLS-1$
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/internal/jface/dialogs/licensing/DiagnosticDialog.java:57]: translatable

File: bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/internal/jface/dialogs/licensing/DiagnosticDialog.java [hunk 5]
```diff
@@ -55,38 +43,48 @@ protected void configureShell(Shell shell) {
...
+				.withColumn("Failure", 50, t -> t.exception().isPresent() ? "Oh, yah..." : "Nope!")//$NON-NLS-1$//$NON-NLS-2$ //$NON-NLS-3$
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.jface/src/org/eclipse/passage/lic/internal/jface/dialogs/licensing/DiagnosticDialog.java:58]: translatable, translatable, translatable

**Pull request merged**